musl-related compile fixes based on openembedded patchset for autofs

https://github.com/openembedded/meta-openembedded/
(see meta-networking/recipes-daemons/autofs/autofs/)

diff -ur autofs-5.1.2/daemon/lookup.c.orig autofs-5.1.2/daemon/lookup.c
--- autofs-5.1.2/daemon/lookup.c.orig	2016-06-15 02:40:44.000000000 +0000
+++ autofs-5.1.2/daemon/lookup.c	2018-05-22 19:37:53.698931707 +0000
@@ -366,7 +366,7 @@
 	if (!S_ISREG(st.st_mode))
 		return NSS_STATUS_NOTFOUND;
 
-	if (st.st_mode & __S_IEXEC)
+	if (st.st_mode & S_IEXEC)
 		type = src_prog;
 	else
 		type = src_file;
@@ -856,7 +856,7 @@
 	if (!S_ISREG(st.st_mode))
 		return NSS_STATUS_NOTFOUND;
 
-	if (st.st_mode & __S_IEXEC)
+	if (st.st_mode & S_IEXEC)
 		type = src_prog;
 	else
 		type = src_file;
diff -ur autofs-5.1.2/include/automount.h.orig autofs-5.1.2/include/automount.h
--- autofs-5.1.2/include/automount.h.orig	2016-06-15 02:40:44.000000000 +0000
+++ autofs-5.1.2/include/automount.h	2018-05-22 19:38:01.242979571 +0000
@@ -42,6 +42,14 @@
 
 #define ENABLE_CORES	1
 
+#ifndef __SWORD_TYPE
+# if __WORDSIZE == 32  /* System word size */
+#  define __SWORD_TYPE int
+# else /* __WORDSIZE == 64 */
+#  define __SWORD_TYPE long int
+# endif
+#endif
+
 /* We MUST have the paths to mount(8) and umount(8) */
 #ifndef HAVE_MOUNT
 #error Failed to locate mount(8)!
diff -ur autofs-5.1.2/include/nsswitch.h.orig autofs-5.1.2/include/nsswitch.h
--- autofs-5.1.2/include/nsswitch.h.orig	2016-06-15 02:40:44.000000000 +0000
+++ autofs-5.1.2/include/nsswitch.h	2018-05-22 19:38:01.242979571 +0000
@@ -24,6 +24,9 @@
 #include <netdb.h>
 #include "list.h"
 
+#ifndef _PATH_NSSWITCH_CONF
+#define _PATH_NSSWITCH_CONF     "/etc/nsswitch.conf"
+#endif
 #define NSSWITCH_FILE _PATH_NSSWITCH_CONF
 
 enum nsswitch_status {
diff -ur autofs-5.1.2/include/rpc_subs.h.orig autofs-5.1.2/include/rpc_subs.h
--- autofs-5.1.2/include/rpc_subs.h.orig	2016-06-15 02:40:44.000000000 +0000
+++ autofs-5.1.2/include/rpc_subs.h	2018-05-22 19:38:10.547038606 +0000
@@ -18,7 +18,7 @@
 
 #include <rpc/rpc.h>
 #include <rpc/pmap_prot.h>
-#include <nfs/nfs.h>
+#include <linux/nfs.h>
 #include <linux/nfs2.h>
 #include <linux/nfs3.h>
 
diff -ur autofs-5.1.2/modules/lookup_multi.c.orig autofs-5.1.2/modules/lookup_multi.c
--- autofs-5.1.2/modules/lookup_multi.c.orig	2016-06-15 02:40:44.000000000 +0000
+++ autofs-5.1.2/modules/lookup_multi.c	2018-05-22 19:38:45.231258767 +0000
@@ -247,7 +247,7 @@
 				continue;
 			}
 
-			if (st.st_mode & __S_IEXEC)
+			if (st.st_mode & S_IEXEC)
 				type = src_prog;
 			else
 				type = src_file;
@@ -452,7 +452,7 @@
 					continue;
 				}
 
-				if (st.st_mode & __S_IEXEC)
+				if (st.st_mode & S_IEXEC)
 					type = src_prog;
 				else
 					type = src_file;
